codalab/lib/upload_manager.py

```diff
@@ -45,77 +46,100 @@ def upload_to_bundle_store(
         |simplify_archives|: whether to simplify unpacked archives so that if they
                              contain a single file, the final path is just that file,
                              not a directory containing that file.
-
+        |is_dir_upload|: whether the incoming sources are individual files that need to be rebuild
+        into a directory or merely individual files.
         If |sources| contains one source, then the bundle contents will be that source.
         Otherwise, the bundle contents will be a directory with each of the sources.
         Exceptions:
         - If |git|, then each source is replaced with the result of running 'git clone |source|'
         - If |unpack| is True or a source is an archive (zip, tar.gz, etc.), then unpack the source.
         """
-        exclude_patterns = (
-            self._default_exclude_patterns + exclude_patterns
-            if exclude_patterns
-            else self._default_exclude_patterns
-        )
-        bundle_path = self._bundle_store.get_bundle_location(bundle.uuid)
-        try:
-            path_util.make_directory(bundle_path)
-            # Note that for uploads with a single source, the directory
-            # structure is simplified at the end.
-            for source in sources:
-                is_url, is_local_path, is_fileobj, filename = self._interpret_source(source)
-                source_output_path = os.path.join(bundle_path, filename)
-                if is_url:
-                    if git:
-                        source_output_path = file_util.strip_git_ext(source_output_path)
-                        file_util.git_clone(source, source_output_path)
-                    else:
-                        file_util.download_url(source, source_output_path)
-                        if unpack and self._can_unpack_file(source_output_path):
+        if is_dir_upload:
```

Comment:
I thought we went with the version where we zip up things on the client side, but we switched back to uploading individual files?